### PR TITLE
[XLA:CPU] Bug fix for enabling F16 support

### PR DESCRIPTION
**Pull request by kanvi-nervana**
This PR fixes a bug for f16 matmul enabling.
Since dot is supported in f16, there is no need for adding an extra convert instruction after fusion similar to the bf16 case.
Also add missed changes in build file.

### CODE REVIEW

File: xla/service/cpu/onednn_matmul_rewriter.cc [hunk 1]
```diff
@@ -520,26 +515,30 @@ class OneDnnMatMulRewriteVisitor : public DfsHloRewriteVisitor {
       // for bf16 case to avoid datatype mismatch.
```
Comment (penpornk) [inline xla/service/cpu/onednn_matmul_rewriter.cc:515]: Please update the comment here as well (that convert for bf16 is optional now).

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

**Comment by kanvi-nervana** [comment]
@penpornk I have addressed the comment.

**Review by penpornk — APPROVED** [review]
Thank you for the change!

**Pull request closed without merging**